src/material/schematics/ng-generate/theming-api/migration.ts

```diff
@@ -0,0 +1,251 @@
+  'mat-light-theme': 'define-light-theme',
+  'mat-typography-level': 'define-typography-level',
+  'mat-typography-config': 'define-typography-config',
+  'mat-font-size': 'font-size',
+  'mat-line-height': 'line-height',
+  'mat-font-weight': 'font-weight',
+  'mat-letter-spacing': 'letter-spacing',
+  'mat-font-family': 'font-family',
+};
+
+/** Mapping of Material variables that should be renamed. */
+const materialVariables: Record<string, string> = {
+  'mat-light-theme-background': 'light-theme-background-palette',
+  'mat-dark-theme-background': 'dark-theme-background-palette',
+  'mat-light-theme-foreground': 'light-theme-foreground-palette',
+  'mat-dark-theme-foreground': 'dark-theme-foreground-palette',
+};
+
+// The palettes all follow the same pattern.
+[
+  'red', 'pink', 'indigo', 'purple', 'deep-purple', 'blue', 'light-blue', 'cyan', 'teal', 'green',
+  'light-green', 'lime', 'yellow', 'amber', 'orange', 'deep-orange', 'brown', 'grey', 'gray',
+  'blue-grey', 'blue-gray'
+].forEach(name => materialVariables[`mat-${name}`] = `${name}-palette`);
+
+/** Mapping of CDK variables that should be renamed. */
+const cdkVariables: Record<string, string> = {
+  'cdk-z-index-overlay-container': 'z-index-overlay-container',
+  'cdk-z-index-overlay': 'z-index-overlay',
+  'cdk-z-index-overlay-backdrop': 'z-index-overlay-backdrop',
+  'cdk-overlay-dark-backdrop-background': 'overlay-dark-backdrop-background',
+};
+
+/** Mapping of CDK mixins that should be renamed. */
+const cdkMixins: Record<string, string> = {
+  'cdk-overlay': 'overlay',
+  'cdk-a11y': 'a11y-visually-hidden',
+  'cdk-high-contrast': 'high-contrast',
+  'cdk-text-field-autofill-color': 'text-field-autofill-color',
+  // This one was split up into two mixins, but which is trickier to
+  // migrate so for now we forward to the deprecated variant.
+  'cdk-text-field': 'text-field',
+};
+
+/**
+ * Migrates the content of a file to the new theming API. Note that this migration is using plain
+ * string manipulation, rather than the AST from PostCSS and the schematics string manipulation
+ * APIs, because it allows us to run it inside g3 and to avoid introducing new dependencies.
+ * @param content Content of the file.
+ * @param oldImportPath Old import to Angular theming API (e.g. `~@angular/material/theming`).
+ * @param newMaterialImportPath New import to the Material theming API (e.g. `~@angular/material`).
+ * @param newCdkImportPath New import to the CDK Sass APIs (e.g. `~@angular/cdk`).
+ */
+export function migrateFileContent(content: string,
+                                   oldImportPath: string,
+                                   newMaterialImportPath: string,
+                                   newCdkImportPath: string): string {
+  const initialContent = content;
+
+  // Replace the old Material import with the new one.
+  content = content
+    .replace(new RegExp(`@import +['"]${escapeRegExp(oldImportPath)}['"].*;\n`, 'g'), '');
```

Comment:
It looks like this only targets places using `@import` and not people who have switch to `@use` already against the old API. Could we handle the `@use` cases too? It is a bit more work, but I think it should still be doable just based on string replacement.

Reply:
So the case where people were using `@use '~@angular/material/theming'`?

Reply:
Ideally we could be a bit more general about the path since inside Google people reference the files deeply into the project. Since we know that we don't have any name collisions, we can assume that any import path containing `angular2_material` is effectively equivalent to `@angular/material/theming`. 

As for what to replace it with- if the script can define constants like below, we can use copybara to transform them into our internal paths
```ts
const cdkUseUrl = '~@angular/cdk';
const materialUseUrl = '~@angular/material';
```

Reply:
It should be doable, but I'll a little iffy because the new Sass entry point doesn't expose all of our mixins. The way I see it is that if somebody is using a deep import path, they're already looking for something that isn't public.